### PR TITLE
Reports: Output summary job

### PR DESCRIPTION
**Pull request by psiinon**
Part of https://github.com/zaproxy/zaproxy/issues/6483

Required to replicate the packaged scan functionality.

No unit tests, but not really sure what useful ones could be added :/
I've been manually testing the new baseline code - thats much easier as you can then compare the output side by side...

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

**Comment by psiinon** [comment]
The other changes have been split out into separate PRs.
But keeping as WIP as I want to see if I can add some unit tests...

**Comment by psiinon** [comment]
Added tests so no ready for review.

**Comment by psiinon** [comment]
Just tweaked to order the rules by String id instead of Int id - for backwards compatibility :)
Have included a test for that too.

**Comment by thc202** [comment]
This seems to overlap with #2967, better merge that one first?

**Comment by psiinon** [comment]
Fixed conflict

File: addOns/reports/src/main/java/org/zaproxy/addon/reports/automation/OutputSummaryJob.java [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+        int pass = 0;
+        int warn = 0;
+
+        // Number of URLs, as per logic behind the core.urls API endpoint
+        int numUrls = ext.countNumberOfUrls();
+        if (numUrls == 0) {
+            out.println(
+                    "No URLs found - is the target URL accessible? Local services may not be accessible from a Docker container");
+
```
Comment (kingthorin) [inline addOns/reports/src/main/java/org/zaproxy/addon/reports/automation/OutputSummaryJob.java:86]: Could be removed.

**Review by kingthorin — COMMENTED** [review]
I don't want to make you redo stuff, but do you think it would be more obvious for later maint etc if the UnitTest output used `\t` instead of tab (whitespace)?

File: addOns/reports/src/main/java/org/zaproxy/addon/reports/automation/OutputSummaryJob.java (hunk 1 above)
Comment (psiinon) [reply addOns/reports/src/main/java/org/zaproxy/addon/reports/automation/OutputSummaryJob.java:86]: No, thats what the baseline says so this is also for backwards compatibility :)
https://github.com/zaproxy/zaproxy/blob/main/docker/zap-baseline.py#L356

**Comment by psiinon** [comment]
No problem using \t - I just pasted the expected output in there ;)

File: addOns/reports/src/main/java/org/zaproxy/addon/reports/automation/OutputSummaryJob.java (hunk 1 above)
Comment (kingthorin) [reply addOns/reports/src/main/java/org/zaproxy/addon/reports/automation/OutputSummaryJob.java:86]: I just meant the blank line 😉

Comment (psiinon) [reply addOns/reports/src/main/java/org/zaproxy/addon/reports/automation/OutputSummaryJob.java:86]: Ha! I didnt even notice it ;)

**Review by kingthorin — APPROVED** [review]

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**